baseplate/diagnostics/tracing.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import calendar
+import json
+import logging
+import socket
+import time
+from datetime import datetime
+
+import pytz
+import requests
+
+from ..core import BaseplateObserver, SpanObserver, TraceInfo
+
+
+logger = logging.getLogger(__name__)
+
+
+# Span annotation types
+ANNOTATIONS = {
+    'CLIENT_SEND': 'cs',
+    'CLIENT_RECEIVE': 'cr',
+    'SERVER_SEND': 'ss',
+    'SERVER_RECEIVE': 'sr',
+}
+
+
+def current_epoch_microseconds():
+    """Returns current UTC time since epoch in microseconds."""
```

Comment:
haha, no, just a stickler for the consistency-enforcing conventions :)

(there's a pep257 tool you can run to check docstrings)